### PR TITLE
Delete single- and multi-part objects in Swift

### PR DESCRIPTION
**Pull request by andrewgaul**
Due to Swift's wonky MPU implementation we must try the multi-part
delete first followed by a single-part delete if it fails.  Previously
delete would leave orphan subobjects.  Fixes andrewgaul/s3proxy#92.

### CODE REVIEW

**Comment by andrewgaul** [comment]
@zack-shoylev This commit makes me sad but I do not see another way around this.  Thoughts?

File: apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java [hunk 1]
```diff
@@ -347,9 +348,20 @@ public Blob getBlob(String container, String name, GetOptions options) {
...
+      // not an MPU we fall back to single-part delete.
+      DeleteStaticLargeObjectResponse response = api.getStaticLargeObjectApi(regionId, container).delete(name);
+      if (!response.status().equals("200 OK")) {
+         api.getObjectApi(regionId, container).delete(name);
+      }
    }
 
+   /**
+    * Delete multiple single-part objects.  Note that this cannot remove the
+    * subobjects of a mulit-part upload.
```
Comment (zack-shoylev) [inline apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java]: Small typo: mulit

**Comment by zack-shoylev** [comment]
It looks fine to me, there shouldn't be an alternative especially considering how removeBlob should work. From what I understand, with the way different providers work with MPU, swift just shifts some of the complexity to dealing with manifests (instead of, say, assembling the MPU objects). Oh well.

File: apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java (hunk 1 above)
Comment (andrewgaul) [inline apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java]: Fixed.

**Pull request merged**